util/ibex_config.py

```diff
@@ -0,0 +1,145 @@
+    if not isinstance(config_dict, collections.Mapping):
+        raise ConfigException('Config ' + name +
+                              ' must have dictionary giving parameters')
+
+    for k, v in config_dict.items():
+        if isinstance(v, int):
+            continue
+        if isinstance(v, str):
+            continue
+
+        raise ConfigException('Parameter ' + k + ' for config ' + name +
+                              ' must be string or int got ' + str(type(v)))
+
+
+def get_config_dicts(config_file):
+    """From a open file config_file extract a dictionary of configuration
+    dictionaries
+
+    Throws ConfigException on any error"""
+    try:
+        config_hjson = hjson.load(config_file)
+
+        for k, v in config_hjson.items():
+            verify_config(k, v)
+
+        verify_config_parameters(config_hjson)
+
+        return config_hjson
+    except hjson.HjsonDecodeError as e:
+        raise ConfigException('Could not decode hjson ' + str(e))
+
+
+def verify_config_parameters(config_dicts):
+    """Verify parameters across config_dicts match expectations:
+        - Each config has the same set of parameters specified"""
+    parameters = set()
+
+    first = True
+
+    for name, config_dict in config_dicts.items():
+        parameters_this_config = set()
+
+        for parameter, value in config_dict.items():
+            if first:
+                parameters.add(parameter)
+
+            parameters_this_config.add(parameter)
+
+        if first:
+            first = False
+        else:
+            parameter_difference = parameters ^ parameters_this_config
+            if len(parameter_difference) != 0:
+                raise ConfigException('Config ' + name +
+                                      ' has differing parameters ' +
+                                      ','.join(parameter_difference))
+
+
+def config_dict_to_fusesoc_opts(config_dict):
+    fusesoc_cmd = []
+    for parameter, value in config_dict.items():
+        fusesoc_cmd.append('--' + parameter + '=' + str(value))
+
+    return ' '.join(fusesoc_cmd)
+
+
+def get_config_file_location():
+    """Returns the location of the config file, IBEX_CONFIG_FILE environment
+    variable overrides the default"""
+
+    if 'IBEX_CONFIG_FILE' in os.environ:
+        return os.environ['IBEX_CONFIG_FILE']
+
+    return DEFAULT_CONFIG_FILE
+
+
+def main():
+    config_outputs = {'fusesoc_opts': config_dict_to_fusesoc_opts}
+
+    argparser = argparse.ArgumentParser(
+        description="""Outputs Ibex configuration
+    parameters for a named config in a number of formats.  If not specified on
+    the command line the config will be read from {0}.  This default can be
+    overridden by setting the IBEX_CONFIG_FILE environment variable
+    """.format(get_config_file_location()))
+
+    argparser.add_argument('config_name',
+                           help="""The name of the Ibex
+    configuration to output""")
```

Comment:
This is rather ugly. Better:

    help=('The name of the Ibex '
          'configuration to output')

(and you only need to wrap at column 80).

Reply:
This was post running it through the pylint script in the OT repo, it does sometimes produce odd outputs, I'll go with your suggestion (no python lint CI to fail so doesn't matter if the linter doesn't agree), after taking a look at the style guide.